extensions/mssql/src/features.ts

```diff
@@ -86,7 +91,40 @@ export class AccountFeature implements StaticFeature {
 
 		let params: contracts.RequestSecurityTokenResponse = {
 			accountKey: JSON.stringify(account.key),
-			token: securityToken.token
+			token: securityToken.token,
+			expiresOn: securityToken.expiresOn
+		};
+
+		return params;
+	}
+
+	protected async refreshToken(request: contracts.RefreshTokenNotificationParams) {
+
+		// find account
+		const accountList = await azdata.accounts.getAllAccounts();
+		let account: azdata.Account;
+		accountList.forEach(element => {
+			if (element.key.accountId === request.accountId) {
+				account = element;
+			}
+		});
+
+		// find tenant
+		const tenant = account.properties.tenants.find((tenant: azurecore.Tenant) => request.authority.includes(tenant.id));
```

Comment:
This is an odd way to get the tenant - is the ID guaranteed to be a part of the authority? Generally requests that involve tokens already have a specific tenant they're operating against so I'm not sure why we need to do this here.